src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCode.cs

```diff
@@ -1,111 +1,37 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-// This RegexCode class is internal to the regular expression package.
-
-// Implementation notes:
-//
-// Regexps are built into RegexCodes, which contain an operation array,
-// a string table, and some constants.
-//
-// Each operation is one of the codes below, followed by the integer
-// operands specified for each op.
-//
-// Strings and sets are indices into a string table.
-
 using System.Collections;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 
 namespace System.Text.RegularExpressions
 {
+    /// <summary>Representation of a regular expression, written by <see cref="RegexWriter"/> and containing the code evaluated by <see cref="RegexInterpreter"/>.</summary>
+    /// <remarks>It currently stores some data used by engines other than the interpreter; that can be refactored out in the future.</remarks>
```

Comment:
I suppose by this you mean mainly the RegexFindOptimizations?

Reply:
Oh and the Tree of course. I guess we could log an issue to extract the Tree out, and have the optimizations be part of the tree object instead since in fact the optimizations are only based on the Tree and not the Codes IIRC.

Reply:
Yeah, in my mind that's part of https://github.com/dotnet/runtime/issues/62450.